frame/contracts/src/wasm/runtime.rs

```diff
@@ -2821,4 +2827,28 @@ pub mod env {
 		ctx.charge_gas(RuntimeCosts::InstantationNonce)?;
 		Ok(ctx.ext.nonce())
 	}
+
+	#[unstable]
+	fn add_delegate_dependency(
+		ctx: _,
+		memory: _,
+		code_hash_ptr: u32,
+	) -> Result<ReturnCode, TrapReason> {
```

Comment:
Function is infallible (diverges on error).

```suggestion
	) -> Result<(), TrapReason> {
```

Reply:
how do you write this when you return the empty tuple?
```wat
	(import "seal0" "add_delegate_dependency" (func $add_delegate_dependency (param i32) (result i32)))
```

Reply:
tests complained and rejected the code with the suggested change
```wat
	(import "seal0" "add_delegate_dependency" (func $add_delegate_dependency (param i32) (result)))
```
does not sound valid

Reply:
You just leave out the `(result)`.

```wat
	(import "seal0" "add_delegate_dependency" (func $add_delegate_dependency (param i32)))
```

Reply:
this does not work as well 
see commits tested here https://github.com/paritytech/substrate/pull/14079/commits/a876168f2054edf84d720c666387583ccbe78dcd

it fails as well with a rejected code error

Reply:
Because you were assuming that there is a return type when calling the function.

Reply:
ha make sense, would be nice to get more detailed error message from the lib here